cluster/addons/dns-horizontal-autoscaler/dns-horizontal-autoscaler.yaml

```diff
@@ -103,4 +103,6 @@ spec:
       tolerations:
       - key: "CriticalAddonsOnly"
         operator: "Exists"
+      nodeSelector:
+        beta.kubernetes.io/os: linux
```

Comment:
Should we use `kubernetes.io/os` instead? seems like the beta one is deprecated [1].

[1] https://kubernetes.io/docs/reference/kubernetes-api/labels-annotations-taints/#beta-kubernetes-io-os-deprecated